### PR TITLE
Feature/Primary button

### PR DESCRIPTION
**Pull request by andgreendigital**
Damn snazzy!

### CODE REVIEW

File: src/components/button.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    icon: PropTypes.oneOf(['arrow', 'plus']),
+    level: PropTypes.oneOf(['primary']),
+    onClick: PropTypes.func,
+};
+
+Button.defaultProps = {
+    type: 'button',
+    icon: 'arrow',
+    level: 'primary',
+    onClick: null,
```
[the remark this answers is not in the document]
Comment (andgreendigital) [reply src/components/button.jsx]: It's considered good practice to give all non-required fields default props for documentation purposes. https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md

File: src/components/button.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Plus from './icons/plus';
+import css from './button.module.css';
+
+const icons = {
+    arrow: <ForwardArrow arrowClassName={css.icon_fill} circleClassName={css.icon_background} />,
+    plus: <Plus plusClassName={css.icon_fill} circleClassName={css.icon_background} />,
+};
+
+const Button = ({
+    text, type, icon, level, onClick,
```
Comment (yosser) [inline src/components/button.jsx]: 'level' is for future proofing purposes I guess?

File: src/components/button.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aberry95) [reply src/components/button.jsx]: not a fan of setting things as null, but that is a preference thing. 
approved

File: src/components/button.jsx (hunk 2 above)
Comment (andgreendigital) [reply src/components/button.jsx]: Indeed! :smirk: Watch this space! 👀

File: src/components/icons/plus.test.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import Plus from './plus';
+
+describe('<Plus/>', () => {
+    it('Icon renders', () => {
+        const component = shallow(<Plus />);
```
Comment (eugene-matvejev) [inline src/components/icons/plus.test.jsx]: it is not as descriptive pattern, as current :)


when it comes to ISO reports, this fits better :)
```
import React from 'react';
import Card from './card';

describe('<Card/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Card {...props} />);

            expect(c).toMatchSnapshot();
        });
    });
});
```

File: src/components/icons/plus.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Plus = ({ plusClassName, circleClassName }) => (
+    <svg style={{ verticalAlign: 'top' }} width="40px" height="40px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
```
Comment (eugene-matvejev) [inline src/components/icons/plus.jsx]: width/heigh would be better to pass as prop

File: src/components/icons/hamburger.test.jsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import Hamburger from './hamburger';
+
+describe('<Hamburger/>', () => {
+    it('Icon renders', () => {
```
Comment (eugene-matvejev) [inline src/components/icons/hamburger.test.jsx]: as comment below :)

File: src/components/icons/hamburger.jsx [hunk 6]
```diff
@@ -1,25 +1,15 @@
...
-    <svg
-        width="24px"
-        height="12px"
-        viewBox="0 0 24 12"
-    >
+    <svg width="24px" height="12px" viewBox="0 0 24 12">
         <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
             <g id="Organisms-/-Elements-/-Header" transform="translate(-1322.000000, -34.000000)" fill="#FF323C">
-                <g id="Atoms-/-icons-/-System-/-Hamburger" transform="translate(1322.000000, 34.000000)">
+                <g transform="translate(1322.000000, 34.000000)">
```
Comment (eugene-matvejev) [inline src/components/icons/hamburger.jsx]: 👍

File: src/components/icons/hamburger.jsx [hunk 7]
```diff
@@ -1,25 +1,15 @@
 import React from 'react';
 
 const Hamburger = () => (
-    <svg
-        width="24px"
-        height="12px"
-        viewBox="0 0 24 12"
-    >
+    <svg width="24px" height="12px" viewBox="0 0 24 12">
```
Comment (eugene-matvejev) [inline src/components/icons/hamburger.jsx]: 👍

File: src/components/icons/forward-arrow.test.jsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ForwardArrow from './forward-arrow';
+
+describe('<ForwardArrow/>', () => {
+    it('Icon renders', () => {
+        const component = shallow(<ForwardArrow />);
```
Comment (eugene-matvejev) [inline src/components/icons/forward-arrow.test.jsx]: as per comment below :)

File: src/components/icons/forward-arrow.jsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ForwardArrow = ({ arrowClassName, circleClassName }) => (
+    <svg style={{ verticalAlign: 'top' }} width="40px" height="40px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
```
Comment (eugene-matvejev) [inline src/components/icons/forward-arrow.jsx]: as per comment below

File: src/components/icons/__snapshots__/hamburger.test.jsx.snap [hunk 10]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`<Hamburger/> render with default/required props 1`] = `
+exports[`<Hamburger/> Icon renders 1`] = `
```
Comment (eugene-matvejev) [inline src/components/icons/__snapshots__/hamburger.test.jsx.snap]: previous report was more descriptive :(

File: src/components/button.jsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        <div>
+            {icons[icon]}
+        </div>
+    </button>
+);
+
+Button.propTypes = {
+    text: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(['button', 'submit', 'reset']),
+    icon: PropTypes.oneOf(['arrow', 'plus']),
```
Comment (eugene-matvejev) [inline src/components/button.jsx]: it is not the best way forward , as it always initiate them

I think we can pass icon as node prop, or refer to sprite, or refer to <Icon/> component with prop, but having them part of a button is a bit too much :)

File: src/components/button.jsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+        </div>
+        <div>
+            {icons[icon]}
+        </div>
+    </button>
+);
+
+Button.propTypes = {
+    text: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(['button', 'submit', 'reset']),
```
Comment (eugene-matvejev) [inline src/components/button.jsx]: i wouldn't bother to fix HTML logic, as it comes outof the box

File: src/components/button.jsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    arrow: <ForwardArrow arrowClassName={css.icon_fill} circleClassName={css.icon_background} />,
+    plus: <Plus plusClassName={css.icon_fill} circleClassName={css.icon_background} />,
+};
+
+const Button = ({
+    text, type, icon, level, onClick,
+}) => (
+    <button type={type} className={`${css.button__container} font_style__B2`} onClick={onClick}>
+        <div className={css.button__text}>
+            { text }
```
Comment (eugene-matvejev) [inline src/components/button.jsx]: if we follow majority of conventions, they name such things as a label not as a text

File: src/components/button.jsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+        </div>
+    </button>
+);
+
+Button.propTypes = {
+    text: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(['button', 'submit', 'reset']),
+    icon: PropTypes.oneOf(['arrow', 'plus']),
+    level: PropTypes.oneOf(['primary']),
+    onClick: PropTypes.func,
```
Comment (eugene-matvejev) [inline src/components/button.jsx]: i wouldn't be bothered to document navite react stuff, just pass them 'as-is' e.g.
const Button = ({ xxx, ...props }) => <button xxx {...props}> ... </button>

File: src/components/icons/plus.jsx (hunk 4 above)
Comment (andgreendigital) [reply src/components/icons/plus.jsx]: Can be added if it is needed! 🙂

File: src/components/button.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (andgreendigital) [reply src/components/button.jsx]: Well its gone now! 😛

File: src/components/icons/__snapshots__/hamburger.test.jsx.snap (hunk 10 above)
Comment (andgreendigital) [reply src/components/icons/__snapshots__/hamburger.test.jsx.snap]: But this icon doesn't have any props!

File: src/components/button.jsx (hunk 12 above)
Comment (andgreendigital) [reply src/components/button.jsx]: I included it so it can have a default as for accessibility you should specify a type.

**Pull request merged**